openedx_events/data.py

```diff
@@ -0,0 +1,62 @@
+"""
+Data attributes for events within the architecture subdomain `learning`.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+"""
+import socket
+from datetime import datetime
+from uuid import UUID, uuid1
+
+import attr
+from django.conf import settings
+
+import openedx_events
+
+
+@attr.s(frozen=True)
+class EventsMetadata:
+    """
+    Attributes defined for Open edX Events metadata object.
+
+    The attributes defined in this class are a subset of the
+    OEP-41: Asynchronous Server Event Message Format.
+
+    Arguments:
+        id (UUID): event identifier.
+        event_type (str): name of the event.
+        minorversion (int): version of the event type.
+        source (str): logical source of an event.
+        sourcehost (str): physical source of the event.
+        time (datetime): timestamp when the event was sent.
+        sourcelib (str): Open edX Events library version.
```

Comment:
@mariajgrimaldi: Hello Maria. It seems we went with a tuple instead of a string. I'm curious why we did that. I'm working on the event bus, which is going to need to send this metadata across the wire, so I may need to pass in a string and convert it back to a tuple, which will be a bit strange, but would make the type in this comment more correct.

Reply:
@robrap Hello there, Robert. After some digging, I found this discussion I had with dave when this PR was open: https://github.com/openedx/openedx-events/pull/7#discussion_r667294019 If you think there's room for improvement, let me know! -besides updating the docstring 😅 -

Reply:
Thank you, Maria. I'm updating in this PR: https://github.com/openedx/openedx-events/pull/168/files#r1083076265